app/helpers/view_components/site_header.rb

```diff
@@ -53,22 +54,27 @@ def signed_in_section
     end
 
     def signed_in_nav
```

Comment:
Let's use Rails Cache to cache the HTML for this so we don't have to render it on every view (as it's going to be expensive).

You can wrap the content of this method in something like:
```
Rails.cache.fetch("top-nav-signed-in") do 
...
end
```

Check if we use any sort of naming schema for Rails cache by looking at other places in the UI